app/src/main/java/de/westnordost/streetcomplete/screens/main/map/LocationAwareMapFragment.kt

```diff
@@ -93,10 +105,23 @@ open class LocationAwareMapFragment : MapFragment() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        displayedLocation = savedInstanceState?.getParcelable(DISPLAYED_LOCATION)
-        val nullTerminatedTracks = savedInstanceState?.getParcelableArrayList<Location?>(TRACKS) as ArrayList<Location?>?
-        if (nullTerminatedTracks != null) {
-            tracks = nullTerminatedTracks.unflattenNullTerminated()
+        // Restore value of members from saved state
+        if (savedInstanceState != null) {
+            with(savedInstanceState) {
+                displayedLocation = getParcelable(DISPLAYED_LOCATION)
+                isRecordingTracks = getBoolean(TRACKS_IS_RECORDING)
+                // It seems that the last list element will be an empty one normally
+                // If it is an empty one we can just remove it so we can keep on recording
+                val nullTerminatedTracks =
+                    getParcelableArrayList<Location?>(TRACKS) as ArrayList<Location?>?
+                if (nullTerminatedTracks != null) {
+                    tracks = nullTerminatedTracks.unflattenNullTerminated()
+                    // unflattenNullTerminated creates an empty list item at the end
+                    if (isRecordingTracks) {
```

Comment:
I suggested to change the condition `tracksRecording && tracks.last().isEmpty()` to `tracks.last().isEmpty()`, not to `tracksRecording`. Sorry if I wasn't clear.

I don't see why the last list item shouldn't be removed when not recording tracks.

Reply:
This might be a preference thing, but if I leave the app and re-enter, it is nice to *not* have the big "jump" in the trajectory and just have the track start recording again since GPS doesn't record in the background. This is why I left it in the non-recording the case. I can change to your suggestion as that makes it more consistent for both cases if you think that is better.

Reply:
Ah I see, so there is a wanted behavior difference. Let's add that to the comment then.

```diff
- // unflattenNullTerminated creates an empty list item at the end
+ // unflattenNullTerminated creates an empty list item (i.e. a new track) at the end.
+ // This is fine if the track is not being recorded.
```